packages/not-next/src/hooks/useClientWithStateManager_test.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { StrictMode } from "react";
+import { assertEquals } from "jsr:@std/assert";
+import { cleanup, renderHook } from "npm:@testing-library/react";
+import { GlobalRegistrator } from "npm:@happy-dom/global-registrator";
+import {
+  createMemoryHistory,
+  createRootRoute,
+  createRoute,
+  createRouter,
+  Outlet,
+  RouterProvider,
+} from "@tanstack/react-router";
+import { createConfig, http, WagmiProvider } from "wagmi";
+import { mainnet, sepolia } from "wagmi/chains";
+import { useClientWithStateManager } from "./useClientWithStateManager.ts";
+
+const config = createConfig({
+  chains: [mainnet, sepolia],
+  transports: {
+    [mainnet.id]: http(),
+    [sepolia.id]: http(),
+  },
+});
+
+const createWrapper = (shopId: string | null = null) => {
+  return ({ children }: { children: React.ReactNode }) => {
+    const rootRoute = createRootRoute({
+      component: Outlet,
+    });
+    const componentRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: "/",
+      component: () => <>{children}</>,
+    });
+    const router = createRouter({
+      routeTree: rootRoute.addChildren([componentRoute]),
+      history: createMemoryHistory({
+        initialEntries: [shopId ? `/?shopId=${shopId}` : "/"],
+      }),
+    });
+
+    return (
+      <StrictMode>
+        <WagmiProvider config={config}>
+          {/* @ts-expect-error */}
```

Comment:
it's expecting `<RouterProvider router={router} />` This is also how we are using it in our `App.tsx`

But if we use that syntax then the router functions don't actually work in testing.